### PR TITLE
Feature/command line in diffirent scenarios

### PR DESCRIPTION
**Pull request by xuzh2024**

### CODE REVIEW

File: hemera/common/enumeration/entity_type.py [hunk 1]
```diff
@@ -125,6 +125,10 @@ def entity_filter_mode(entity_types):
         return False
 
 
+ALL_ENTITY_COLLECTIONS = EntityType.__members__.keys()
```
Comment (shanshuo0918) [inline hemera/common/enumeration/entity_type.py]: Where do you use these？

File: hemera/resource/template/custom_module.example [hunk 2]
```diff
@@ -0,0 +1,54 @@
+from typing import Type, Union
+
+from sqlalchemy import Column, Index, desc, func, text
+from sqlalchemy.dialects.postgresql import BIGINT, BOOLEAN, BYTEA, NUMERIC, TIMESTAMP
+
+from hemera.common.models import HemeraModel, general_converter
+from hemera_udf.${job}.domains import CustomDomainA, CustomDomainB
+
+
+class CustomTable(HemeraModel):
```
Comment (shanshuo0918) [inline hemera/resource/template/custom_module.example]: Provide 3 models/domains
1. AddressFeatureValueCurrent
- address as primary key
- feature_value 
- block_number
- block_timestamp
- create_time
- update_time
3. AddressFeatureValueHistory
- address as primary key
- feature_value 
- block_number as primary key
- block_timestamp
- create_time
- update_time
5. AddressFeatureEvents
- block_number as primary key
- block_timestamp
- transaction_hash as primary key
- address (add a index)
- event_value 
- create_time
- update_time

File: hemera/resource/template/export_custom_job.example [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+class Export${job_name}Job(ExtensionJob):
+    # If the data processing logic of the developed job supports re-running with on-chain data reorg,
+    # set able_to_reorg to True so that the scheduler can automatically start the job during the reorg process.
+    able_to_reorg = False
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def _udf(self, dependency_domains: List[DependencyDomain], output: Collector[Union[CustomDomainA, CustomDomainB]]):
```
Comment (shanshuo0918) [inline hemera/resource/template/export_custom_job.example]: transactions: List[Transactions],
logs: List[Logs],

File: hemera/resource/template/export_custom_job.example [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        job_custom_domain_a_list, job_custom_domain_b_list = some_work_here(dependency_domains)
+
+        # Before collecting, you may need sort/distinct etc.
```
Comment (shanshuo0918) [inline hemera/resource/template/export_custom_job.example]: delete this section

File: hemera/resource/template/export_custom_job.example [hunk 5]
```diff
@@ -0,0 +1,42 @@
+from typing import List, Union
+
+from hemera.indexer.jobs.base_job import ExtensionJob, Collector
+from hemera.indexer.utils.collection_utils import distinct_collections_by_group
+from hemera_udf.${job}.domains import *
+
+
+class Export${job_name}Job(ExtensionJob):
```
Comment (shanshuo0918) [inline hemera/resource/template/export_custom_job.example:8]: Add a filter example

File: hemera/common/enumeration/entity_type.py (hunk 1 above)
Comment (xuzh2024) [reply hemera/common/enumeration/entity_type.py]: 'ALL_ENTITY_COLLECTIONS' will be used in the reorg module.
This modification here is to fix the compatibility errors caused by the refactoring, The actual logic repairs should be submitted separately.

File: hemera/resource/template/custom_module.example (hunk 2 above)
Comment (xuzh2024) [reply hemera/resource/template/custom_module.example]: done

Comment (xuzh2024) [reply hemera/resource/template/custom_module.example]: done

File: hemera/resource/template/export_custom_job.example (hunk 3 above)
Comment (xuzh2024) [reply hemera/resource/template/export_custom_job.example]: done

Comment (xuzh2024) [reply hemera/resource/template/export_custom_job.example]: done

File: hemera/resource/template/export_custom_job.example (hunk 4 above)
Comment (xuzh2024) [reply hemera/resource/template/export_custom_job.example]: done

File: hemera/resource/template/export_custom_job.example (hunk 5 above)
Comment (xuzh2024) [reply hemera/resource/template/export_custom_job.example:8]: done

File: hemera/common/enumeration/entity_type.py (hunk 1 above)
Comment (xuzh2024) [reply hemera/common/enumeration/entity_type.py]: 'ALL_ENTITY_COLLECTIONS' has been removed in the latest commit. 
Now the value will be obtained using DynamicEntityTypeRegistry.

File: hemera/resource/template/custom_module.example [hunk 6]
```diff
@@ -4,15 +4,39 @@ from sqlalchemy import Column, Index, desc, func, text
...
+from hemera_udf.${job}.domains import AddressFeatureValueCurrent, AddressFeatureValueHistory, AddressFeatureEvent
 
 
-class CustomTable(HemeraModel):
-    __tablename__ = "custom_table"
-    primary_key = Column(BYTEA, primary_key=True)
-    number = Column(BIGINT)
-    timestamp = Column(TIMESTAMP)
-    nonce = Column(BYTEA)
+class AddressFeatureValueCurrents(HemeraModel):
```
Comment (shanshuo0918) [inline hemera/resource/template/custom_module.example]: AddressFeatureValuesCurrent?

File: hemera/resource/template/custom_domain.example [hunk 7]
```diff
@@ -4,16 +4,23 @@ from hemera.indexer.domains import Domain
...
 @dataclass
-class CustomDomainB(Domain):
-    field1: str
-    field2: int
-    may_need_block_number: int
-    may_need_block_timestamp: int
\ No newline at end of file
+class AddressFeatureValueHistory(Domain):
+    address: str
+    feature: int
```
Comment (shanshuo0918) [inline hemera/resource/template/custom_domain.example]: feature_value

File: hemera/resource/template/custom_module.example [hunk 8]
```diff
@@ -4,15 +4,39 @@ from sqlalchemy import Column, Index, desc, func, text
...
+            {
+                "domain": AddressFeatureValueCurrent,
+                "conflict_do_update": True,
+                "update_strategy": "EXCLUDED.block_number > address_feature_value_currents.block_number",
+                "converter": general_converter,
+            },
+        ]
+
+
+class AddressFeatureValueHistories(HemeraModel):
```
Comment (shanshuo0918) [inline hemera/resource/template/custom_module.example]: AddressFeatureValuesHistory?

File: hemera/resource/template/custom_module.example [hunk 9]
```diff
@@ -21,27 +45,50 @@ class CustomTable(HemeraModel):
...
-                "conflict_do_update": True,
-                "update_strategy": "EXCLUDED.update_condition_key > custom_table.update_condition_key",
                 "converter": general_converter,
-            },
+            }
         ]
 
 
-Index("custom_table_some_key_index", CustomTable.some_key)
+class AddressFeatureValueEvents(HemeraModel):
```
Comment (shanshuo0918) [inline hemera/resource/template/custom_module.example]: AddressFeatureEvents

**Review by shanshuo0918 — APPROVED** [review]

**Pull request merged**